File: dsa3.js
const prompt = require("prompt-sync")();

// print n hellos

// function printHellos(n) {
//     for (let i = 1; i <= n; i++) {
//         console.log('hello!')
//     }
// }

// let n = Number(prompt('enter a number: '))
// printHellos(n)

// one to n and vice versa

// function oneToNAndBack(n) {
//   for (let i = 1; i <= n; i++) {
//     console.log(i);
//   }
//   for (let i = n-1; i >= 1; i--) {
//     console.log(i);
//   }
// }

// let n = Number(prompt("enter a number: "));
// oneToNAndBack(n);


// sum of natural numbers

// function sumOfNaturalNums(n){
//     let sum = 0;
//     for(let i = 1; i <=n; i++){
//         sum += i
//     }
//     return sum
// }

// let n = Number(prompt("enter a number: "));
// console.log(sumOfNaturalNums(n))

// factorial of a n numbers

// function factorial(n){
//     let ans = 1
//     for (let i = 1; i <= n; i++) {
//         ans *= i
//     }
//     return ans
// }

// let n = Number(prompt("enter a number: "));
// console.log(factorial(n))


// factors of a number n

// function getFactors(n) {
//     let factors = []
//     for(let i = 1; i <= n; i++){
//         if(n%i == 0){
//             factors.push(i)
//         }
//     }
//     return factors
// }

// const n = Number(prompt("enter a number to get their factors: "))
// console.log(getFactors(n))


// prime numbers 


function getFactors(n) {
    let factors = []
    for(let i = 2; i < n; i++){
        if(n%i == 0){
            factors.push(i)
        }
    }
    return factors
}

const n = Number(prompt("enter a number to get their factors: "))
let factors = getFactors(n)

if(factors.length == 0){
    console.log(`${n} is a prime number`)
}else{
    console.log(`${n} is not a prime number`)
}

File: dsa3.py
# printing n hellos


# n = int(input("enter a number: "))

# def printing_n_hellos(n):
#     for i in range(n):
#         print("hello")
        
# printing_n_hellos(n)



# n = int(input("enter a number: "))

# def oneToNAndBack(n):
#     for i in range(1,n+1):
#         print(i)
#     for i in range(n,0,-1):
#         print(i)

# oneToNAndBack(n)



# def sum_of_natural_nums(n):
#     total = 0
#     for i in range(1, n + 1):
#         total += i
#     return total

# n = int(input("Enter a number: "))
# print(sum_of_natural_nums(n))




# def factorial(n):
#     ans = 1
#     for i in range(1,n+1):
#         ans *= i
#     return ans

# n = int(input("enter a number: "))
# print(factorial(n))



# def getFactors(n):
#     factors = []
#     for i in range(1, n+1):
#         if n%i == 0:
#             factors.append(i)
#     return factors


# n = int(input("enter a number to get their factors: "))
# print(getFactors(n))



#  prime number or not



n = int(input("Enter a number: "))

if n <= 1:
    print("Not a prime number")
else:
    is_prime = True

    for i in range(2, int(n**0.5) + 1):
        if n % i == 0:
            is_prime = False
            break

    if is_prime:
        print("Prime number")
    else:
        print("Not a prime number")
